### PR TITLE
resource/aws_efs_file_system: Add support for encryption

### PR DESCRIPTION
**Pull request by pjfbashton**
```
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSEFSFileSystem_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -run=TestAccAWSEFSFileSystem_ -timeout 120m
=== RUN   TestAccAWSEFSFileSystem_importBasic
--- PASS: TestAccAWSEFSFileSystem_importBasic (50.69s)
=== RUN   TestAccAWSEFSFileSystem_basic
--- PASS: TestAccAWSEFSFileSystem_basic (113.40s)
=== RUN   TestAccAWSEFSFileSystem_pagedTags
--- PASS: TestAccAWSEFSFileSystem_pagedTags (48.26s)
=== RUN   TestAccAWSEFSFileSystem_KmsKey
--- PASS: TestAccAWSEFSFileSystem_KmsKey (87.40s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	299.772s
```

### CODE REVIEW

File: aws/resource_aws_efs_file_system.go [hunk 1]
```diff
@@ -76,6 +91,14 @@ func resourceAwsEfsFileSystemCreate(d *schema.ResourceData, meta interface{}) er
 		createOpts.PerformanceMode = aws.String(v.(string))
 	}
 
+	if v, ok := d.GetOk("encrypted"); ok {
+		createOpts.Encrypted = aws.Bool(v.(bool))
+	}
+
+	if v, ok := d.GetOk("kms_key_id"); ok {
+		createOpts.KmsKeyId = aws.String(v.(string))
+	}
```
Comment (Ninir) [inline aws/resource_aws_efs_file_system.go]: We should check that `Encrypted` is true if `KmsKeyId` is set, and catch the related Exception if the API call is made (BadRequest perhaps?).

File: aws/resource_aws_efs_file_system_test.go [hunk 2]
```diff
@@ -170,6 +171,28 @@ func TestAccAWSEFSFileSystem_pagedTags(t *testing.T) {
...
+		CheckDestroy: testAccCheckEfsFileSystemDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAWSEFSFileSystemConfigWithKmsKey(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestMatchResourceAttr(
+						"aws_efs_file_system.foo-with-kms",
+						"kms_key_id",
+						keyRegex,
+					),
```
Comment (Ninir) [inline aws/resource_aws_efs_file_system_test.go]: We should also check that `encrypted` is well set here :)

File: aws/resource_aws_efs_file_system_test.go [hunk 3]
```diff
@@ -351,3 +374,33 @@ resource "aws_efs_file_system" "foo-with-performance-mode" {
...
+	      "Action": "kms:*",
+	      "Resource": "*"
+	    }
+	  ]
+	}
+	POLICY
+	}
+
+	resource "aws_efs_file_system" "foo-with-kms" {
+		encrypted = true
```
Comment (Ninir) [inline aws/resource_aws_efs_file_system_test.go]: `encrypted` should be aligned with `kms_key_id`

File: aws/resource_aws_efs_file_system_test.go (hunk 2 above)
Comment (pjfbashton) [inline aws/resource_aws_efs_file_system_test.go]: Thanks, good call. Check added

File: aws/resource_aws_efs_file_system_test.go [hunk 4]
```diff
@@ -351,3 +374,33 @@ resource "aws_efs_file_system" "foo-with-performance-mode" {
 	performance_mode = "maxIO"
 }
 `
+
+func testAccAWSEFSFileSystemConfigWithKmsKey(rInt int) string {
+	return fmt.Sprintf(`
+	resource "aws_kms_key" "foo" {
```
Comment (pjfbashton) [inline aws/resource_aws_efs_file_system_test.go]: I actually dont think we need that policy at all, test runs fine without it so I removed it

File: aws/resource_aws_efs_file_system.go (hunk 1 above)
Comment (pjfbashton) [inline aws/resource_aws_efs_file_system.go]: Looking into this now, thanks ;)

**Comment by Ninir** [comment]
Ping me when you're all ok on that @pjfbashton, will make the review. Thanks for your efforts :)

**Comment by pjfbashton** [comment]
@Ninir seems that `KmsKeyId` is ignored by sdk when `Encrypted` is false so I added a check that will throw an error if `encrypted` is not `true` when  `kms_key_id` is specified. would that work?

**Comment by pjfbashton** [comment]
That's great @Ninir ! Thanks for your help :+1: ;)

**Comment by shezaan** [comment]
Stupid question, which version of TF will this be added to? I am already on 0.10.4.
Documentation still doesn't reflect this if its already there.
Thanks in advance!

**Comment by Ninir** [comment]
Hi @shezaan !

Since the [Terraform split](https://www.hashicorp.com/blog/upcoming-provider-changes-in-terraform-0-10/), providers (AWS, Google, Azure, Kubernetes, etc...) provide their own releases.

The one including EFS encryption will then be in the next AWS release. I can't provide any ETA of it yet, sadly.
I will try to get back to you on that as soon as possible! 👍

**Pull request merged**